src/uneditableTextField/index.tsx

```diff
@@ -19,20 +19,25 @@ export const UneditableTextField = ({
   ellipsis = false,
   href,
 }: UneditableTextFieldProps) => {
-  let Component: tag
-  let props = {}
+  let componentType: tag
+  let props: any = {}
   if (href && typeof href !== 'string') {
-    Component = href.type
-    props = { ...href.props }
+    componentType = href.type
+    props = href.props
   } else if (typeof href === 'string' && !isEmpty(href)) {
-    Component = 'a'
+    componentType = 'a'
     props = { href }
   } else {
-    Component = 'div'
+    componentType = 'div'
   }
 
-  return (
-    <Component className={cc(['uneditableTextField', className])} {...props}>
+  return React.createElement(
```

Comment:
Btw, I moved the "Component" tag into a React.createElement, since this is recommended, and compliant to Typescript